src/types/app-client.ts

```diff
@@ -1170,29 +1171,29 @@ export class AppClient {
       update: async (params?: AppClientBareCallParams & AppClientCompilationParams & SendParams) => {
         const compiled = await this.compile(params)
         return {
-          ...(await this.handleCallErrors(async () => this._algorand.send.appUpdate(await this.params.bare.update(params)))),
+          ...(await this._algorand.send.appUpdate(await this.params.bare.update(params))),
           ...(compiled as Partial<AppCompilationResult>),
         }
       },
       /** Signs and sends an opt-in call */
       optIn: (params?: AppClientBareCallParams & SendParams) => {
-        return this.handleCallErrors(() => this._algorand.send.appCall(this.params.bare.optIn(params)))
+        return this._algorand.send.appCall(this.params.bare.optIn(params))
```

Comment:
This is a nice side-effect of the change

Reply:
Yeah in general I want to look at features in the high level abstractions (i.e. generated client and app client) and see what we can do at a lower level to better support it and reduce the maintenance burden (and potential feature divergence) on these higher level classes.